src/System.Security.Cryptography.Encoding/src/System/Security/Cryptography/Base64Transforms.cs

```diff
@@ -96,143 +112,224 @@ public class FromBase64Transform : ICryptoTransform
+
+            ReturnToCryptoPool(tmpBufferArray, tmpBuffer.Length);
 
             // reinitialize the transform
             Reset();
 
-            return result;
+            return output;
         }
 
-        private byte[] GetTempBuffer(byte[] inputBuffer, int inputOffset, int inputCount, out int effectiveCount)
+        private Span<byte> GetTempBuffer(Span<byte> inputBuffer, Span<byte> tmpBuffer)
         {
-            byte[] temp;
-
-            if (_whitespaces == FromBase64TransformMode.IgnoreWhiteSpaces)
-            {
-                temp = DiscardWhiteSpaces(inputBuffer, inputOffset, inputCount);
-                effectiveCount = temp.Length;
-            }
-            else
+            if (_whitespaces == FromBase64TransformMode.DoNotIgnoreWhiteSpaces)
             {
-                temp = new byte[inputCount];
-                Buffer.BlockCopy(inputBuffer, inputOffset, temp, 0, inputCount);
-                effectiveCount = inputCount;
+                return inputBuffer;
             }
 
-            return temp;
+            return DiscardWhiteSpaces(inputBuffer, tmpBuffer);
         }
 
-        private byte[] ConvertFromBase64(byte[] temp, int effectiveCount)
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static Span<byte> DiscardWhiteSpaces(Span<byte> inputBuffer, Span<byte> tmpBuffer)
         {
-            // Get the number of 4 bytes blocks to transform
-            int numBlocks = (effectiveCount + _inputIndex) / 4;
+            int count = 0;
 
-            byte[] transformBuffer = new byte[_inputIndex + effectiveCount];
-            Buffer.BlockCopy(_inputBuffer, 0, transformBuffer, 0, _inputIndex);
-            Buffer.BlockCopy(temp, 0, transformBuffer, _inputIndex, effectiveCount);
+            for (int i = 0; i < inputBuffer.Length; i++)
+            {
+                if (!IsWhitespace(inputBuffer[i]))
+                {
+                    tmpBuffer[count++] = inputBuffer[i];
+                }
+            }
 
-            _inputIndex = (effectiveCount + _inputIndex) % 4;
-            Buffer.BlockCopy(temp, effectiveCount - _inputIndex, _inputBuffer, 0, _inputIndex);
+            return tmpBuffer.Slice(0, count);
+        }
 
-            char[] tempChar = Encoding.ASCII.GetChars(transformBuffer, 0, 4 * numBlocks);
-            byte[] tempBytes = Convert.FromBase64CharArray(tempChar, 0, 4 * numBlocks);
-            return tempBytes;
+        private static bool IsWhitespace(byte value)
+        {
+            // We assume ASCII encoded data. If there is any non-ASCII char, so it is invalid
+            // Base64 and will be caught during decoding.
+
+            // SPACE        32
+            // TAB           9
+            // LF           10
+            // VTAB         11
+            // FORM FEED    12
+            // CR           13
+
+            // RyuJIT produces better code with the ternary. This JIT-bug is tracked in https://github.com/dotnet/coreclr/issues/914
```

Comment:
And it has a measurable impact on the public APIs exposed?  If not, we shouldn't clutter the code with it.

Reply:
> measurable impact on the public APIs exposed

It's a minor improvement, just above the noise-level (and depending on the input).
So I'll remove the ternary.

```diff
G_M14072_IG16:
       4D63D1               movsxd   r10, r9d
       460FB61417           movzx    r10, byte  ptr [rdi+r10]
       458BDA               mov      r11d, r10d
       4183FB20             cmp      r11d, 32
       7421                 je       SHORT G_M14072_IG17
       4183C3F7             add      r11d, -9
       4183FB04             cmp      r11d, 4
-      7617                 jbe      SHORT G_M14072_IG17
+      410F96C3             setbe    r11b
+      450FB6DB             movzx    r11, r11b
+      4585DB               test     r11d, r11d
+      7517                 jne      SHORT G_M14072_IG17
       458D5801             lea      r11d, [r8+1]
       443BC1               cmp      r8d, ecx
       0F83C8010000         jae      G_M14072_IG40
       4D63C0               movsxd   r8, r8d
       46881402             mov      byte  ptr [rdx+r8], r10b
       458BC3               mov      r8d, r11d

G_M14072_IG17:
```

Reply:
The comment can be deleted now that the ternary is not present.

Reply:
Ups, missed that. Thanks.